scripts/verify_adv_IM.py

```diff
@@ -32,79 +66,166 @@ def parse_args():
+                df_model["component"] == comp
+            )
+            df_model.loc[comp_mask, "status"] = station_component_status
+            df_model.loc[comp_mask, time_type.start_time.name] = time_list[
+                time_type.start_time.value
+            ]
+            df_model.loc[comp_mask, time_type.end_time.name] = time_list[
+                time_type.end_time.value
+            ]
+    #    print(f"{df_model}")
+    return df_model
 
-def main(im_calc_dir, adv_im_model, components):
 
-    failed = []
+def main(im_calc_dir, adv_im_model, components, simple_check=False, station_file=None):
 
+    df_list = []
     for model in adv_im_model:
         csv_path = os.path.join(im_calc_dir, "{}.csv".format(model))
-        # using try/except to prevent crash
-        try:
-            df = pd.read_csv(csv_path)
-        except FileNotFoundError:
-            failed.append((model, "NoCSV", None))
-            print("csv for {} does not exist".format(model))
-            continue
-        # check for null/nan
-        if df.isnull().values.any():
-            failed.append(
-                (model, "NullValue", df[df.isnull().values].station.to_list())
-            )
 
-        # check station count, if match, skip rest of checks
-        csv_stations = df.station.unique()
-        # glob for station folders
-        list_folders = [
-            y
-            for y in [os.path.join(im_calc_dir, x) for x in os.listdir(im_calc_dir)]
-            if os.path.isdir(y)
-        ]
-        if len(csv_stations) == len(list_folders):
-            # station folder count matches csv.station.count
-            continue
+        if station_file is not None:
+            station_list = load_station_file(station_file).index.tolist()
+        else:
+            # glob for station folders
+            # station_list = [ y for y in [os.path.join(im_calc_dir, x) for x in os.listdir(im_calc_dir)] if os.path.isdir(y) ]
+            station_list = [
+                x
+                for x in os.listdir(im_calc_dir)
+                if os.path.isdir(os.path.join(im_calc_dir, x))
+            ]
+        list_folders = [os.path.join(im_calc_dir, x) for x in station_list]
+
+        # initialize df with empty value with not started
+        df_model = pd.concat(
+            [
+                pd.DataFrame(
+                    {
+                        "station": station_list,
+                        "model": model,
+                        "component": component,
+                        "status": run_status.not_started.value,
+                    },
+                    columns=COLUMN_NAMES,
+                )
+                for component in components
+            ],
+            ignore_index=True,
+        )
+
+        # a quick check to compare station count, will skip all other checks if successful.
+        if simple_check:
+            # using try/except to prevent crash
+            try:
```

Comment:
If we're only checking for filenotfound then we're probably better off using an if rather than a full on try/except? Thoughts?

Reply:
using try/except to read the file prevents the very-unlikely-event that the file got removed (or IO failed) after the isfile check.